homeassistant/helpers/selector.py

```diff
@@ -131,7 +131,20 @@ def _validate_supported_features(supported_features: int | list[str]) -> int:
     return feature_mask
 
 
-ENTITY_FILTER_SELECTOR_CONFIG_SCHEMA = vol.Schema(
+BASE_SELECTOR_CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Optional("read_only"): bool,
+    }
+)
+
+
+class BaseSelectorConfig(TypedDict, total=False):
+    """Class to common options of all selectors."""
+
+    read_only: bool
+
+
```

Comment:
Needs to be added to all the selectors I guess, not only the few below already done.

Reply:
Yes this was limited to just the requirements of this specific PR. I can add the rest in a follow-up?

Reply:
No, I think we should add all in the same PR